src/components/Wizard/Patterns/WizardPattern.js

```diff
@@ -0,0 +1,201 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop, Wizard, Icon, Button } from '../../../index';
+import WizardPatternBody from './WizardPatternBody';
+import { wizardStepShape } from './WizardPatternConstants';
+
+/**
+ * WizardPattern - the Wizard Pattern Body component.
+ */
+const WizardPattern = ({
+  steps,
+  activeStepIndex,
+  onStepChanged,
+  onNext,
+  onBack,
+  nextStepDisabled,
+  title,
+  loadingTitle,
+  loadingMessage,
+  show,
+  onHide,
+  onExited,
+  stepButtonsDisabled,
+  cancelText,
+  backText,
+  nextText,
+  closeText,
+  loading,
+  nextButtonRef,
+  bodyHeader,
+  children
+}) => {
+  const onFirstStep = activeStepIndex === 0;
+  const onFinalStep = activeStepIndex === steps.length - 1;
+
+  const onHideClick = () => {
+    onHide(onFinalStep);
+  };
+
+  const onBackClick = () => {
+    goToStep(Math.max(activeStepIndex - 1, 0));
+  };
+
+  const onNextClick = () => {
+    goToStep(Math.min(activeStepIndex + 1, steps.length - 1));
+  };
+
+  const getStep = (index = activeStepIndex) => steps[index];
+
+  const getPrevStep = (relativeToIndex = activeStepIndex) =>
+    relativeToIndex > 0 && steps[relativeToIndex - 1];
+
+  const getNextStep = (relativeToIndex = activeStepIndex) =>
+    relativeToIndex < steps.length - 1 && steps[relativeToIndex + 1];
+
+  const activeStep = getStep();
+
+  const goToStep = newStepIndex => {
+    if (shouldPreventGoToStep(newStepIndex)) return;
```

Comment:
after looking at this API consumption further downstream, I think the `goToStep` portion needs to ensure all calls can be asynchronous (i.e. any method expecting a boolean is probably better served as a prop coming down, particularly "nextStep" and "previousStep"). One would typically expose a callback on a click, allow the consumer to make async calls, and then set a prop coming down to indicate this. For that reason, I am a little leery on this portion still. It can be made functional, but I don't see this as the "React" way of handling this piece. 

thoughts @mturley ? maybe some of those callbacks can be greatly simplified by passing props from the consumer. I will keep investigating this with you as you resolve downstream issues associated.

Reply:
the other items in this PR from an abstraction standpoint seem OK to me... it's just a matter of this API consumption now...

Reply:
i really do appreciate all the effort on this!!

Reply:
> One would typically expose a callback on a click, allow the consumer to make async calls, and then set a prop (on some step or object or other prop coming down) to indicate this.

:+1: makes perfect sense.

> maybe some of those callbacks can be greatly simplified by passing props from the consumer

I guess this comes down to recognizing what's an actual callback and what's not.

Reply:
ya this is not clear here to me... it seems that `isValid` is being used downstream on the step. I'm OK w/ this, but will wait for @mturley to clarify.